ouroboros-consensus/test-consensus/Test/Dynamic/Network.hs

```diff
@@ -178,6 +187,21 @@ createNetworkInterface chans nodeIds us
         (waitAnyCancel ts `onException` traverse_ cancel ts) >>= k . fst
     }
 
+data TraceChainEvent c ext
+  = SwitchedTo (AnchoredFragment (Header (SimpleBlock c ext)))
+  | Forged (SimpleBlock c ext)
+
+data ChainsTrace c ext = ChainsTrace
+    { ctFinalChains :: Map NodeId
+        ( NodeConfig (BlockProtocol (SimpleBlock c ext))
+        , Chain (SimpleBlock c ext)
+        )
+      -- ^ Each node's final chain.
+    , ctSwitches ::
+        [(CoreNodeId, SlotNo, TraceChainEvent c ext)]
+      -- ^ A record of how each node's chain changed over time.
```

Comment:
Document that these are from new to old.

Maybe add a function that takes care of the reversing and which can be used instead of this accessor. It can be used in the implementation of `prettySwitches`.

Reply:
Done -- I isolate it in a closure.